platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/topology/StaticTopology.java

```diff
@@ -17,78 +17,64 @@
 package com.swirlds.platform.network.topology;
 
 import com.swirlds.common.platform.NodeId;
+import com.swirlds.platform.network.PeerInfo;
 import com.swirlds.platform.network.RandomGraph;
-import com.swirlds.platform.system.address.AddressBook;
 import edu.umd.cs.findbugs.annotations.NonNull;
-import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
-import java.util.Objects;
+import java.util.Map;
 import java.util.Random;
-import java.util.function.Predicate;
+import java.util.Set;
 
 /**
  * A bidirectional topology that never changes.
  */
 public class StaticTopology implements NetworkTopology {
     private static final long SEED = 0;
 
-    private final NodeId selfId;
+    /** nodes are mapped so lookups are efficient. **/
+    private Map<NodeId, Long> peerNodeToIdMap = new HashMap<>();
+
     /**
-     * Two nodes are neighbors if their indexes in the address book are neighbors in the connection graph.
+     * Two nodes are neighbors if their node indexes are neighbors in the connection graph.
      */
-    private final AddressBook addressBook;
-
     private final RandomGraph connectionGraph;
 
+    private final NodeId selfId;
+
     /**
      * Constructor.
      *
      * @param random            a source of randomness, used to chose random neighbors, does not need to be
      *                          cryptographically secure
-     * @param addressBook       the current address book
+     * @param peers             the set of peers in the network
      * @param selfId            the ID of this node
      * @param numberOfNeighbors the number of neighbors each node should have
      */
     public StaticTopology(
             @NonNull final Random random,
-            @NonNull final AddressBook addressBook,
+            @NonNull final List<PeerInfo> peers,
```

Comment:
Is there an invariant on this list that the PeerInfo is in monotonically ascending order with respect to NodeId?

Reply:
yes there is an invariant, but it's that PeerInfo are ordered as they appear in the original addressBook. I understand they should be monotonically ascending as defined in the AB, but I don't know if that's mandated anywhere.

Reply:
outdated, resolving.